layer/tar_writer_factory.go

```diff
@@ -0,0 +1,31 @@
+package layer
+
+import (
+	"archive/tar"
+	"io"
+
+	"github.com/buildpacks/imgutil"
+	"github.com/buildpacks/imgutil/archive"
+)
+
+type tarWriterFactory struct {
+	os string
+}
+
+func NewTarWriterFactory(image imgutil.Image) (archive.TarWriterFactory, error) {
```

Comment:
I think we should return the concrete struct here so we can return structs and accept interfaces when we don't have a compelling reason not to.

We can export `layer.TarWriterFactory` and it's field.

Reply:
Good point. Will look into